tests/test_batchnorm.py

```diff
@@ -14,50 +14,86 @@ def setUp(self):
 
   def test_forward(self) -> None:
 
-    batch_size, channels_in, img_height, img_width = (32, 3, 28, 28)
+    def run_test(input_batchnorm, input_shape, mean, variance, scale, offset, variance_epsilon, channels_first):
 
-    input_shape = [batch_size, channels_in, img_height, img_width]
-    input_batchnorm = np.random.normal(size=input_shape).astype(np.float32)
+      with tfe.protocol.Pond() as prot:
+        batchnorm_input = prot.define_private_variable(input_batchnorm)
 
-    # I reshaped the input because tf.nn.batch_normalization doesn't reshape it
-    # automatically However tf encrypted will reshape automatically the input
-    mean = np.array([2.0, 1.5, 20.8]).reshape(
-        1, channels_in, 1, 1).astype(np.float32)
-    variance = np.array([0.5, 0.3, 0.1]).reshape(
-        1, channels_in, 1, 1).astype(np.float32)
-    scale = np.array([0.3, 0.5, 0.8]).reshape(
-        1, channels_in, 1, 1).astype(np.float32)
-    offset = np.array([1.5, 1.2, 1.4]).reshape(
-        1, channels_in, 1, 1).astype(np.float32)
-    variance_epsilon = 1e-8
+        batchnorm_layer = Batchnorm(input_shape, mean, variance, scale, offset, channels_first=channels_first)
+        batchnorm_layer.initialize()
+        batchnorm_out_pond = batchnorm_layer.forward(batchnorm_input)
 
-    with tfe.protocol.Pond() as prot:
-      batchnorm_input = prot.define_private_variable(input_batchnorm)
+        with tfe.Session() as sess:
+          sess.run(tf.global_variables_initializer())
+          out_pond = sess.run(batchnorm_out_pond.reveal())
 
-      batchnorm_layer = Batchnorm(input_shape, mean, variance, scale, offset)
-      batchnorm_layer.initialize()
-      batchnorm_out_pond = batchnorm_layer.forward(batchnorm_input)
+        # reset graph
+        tf.reset_default_graph()
 
-      with tfe.Session() as sess:
-        sess.run(tf.global_variables_initializer())
-        out_pond = sess.run(batchnorm_out_pond.reveal())
+        with tf.Session() as sess:
+          x = tf.Variable(input_batchnorm, dtype=tf.float32)
 
-      # reset graph
-      tf.reset_default_graph()
-
-      with tf.Session() as sess:
-        x = tf.Variable(input_batchnorm, dtype=tf.float32)
-
-        batchnorm_out_tf = tf.nn.batch_normalization(
+          batchnorm_out_tf = tf.nn.batch_normalization(
             x, mean, variance, offset, scale, variance_epsilon)
 
-        sess.run(tf.global_variables_initializer())
+          sess.run(tf.global_variables_initializer())
 
-        out_tensorflow = sess.run(batchnorm_out_tf)
+          out_tensorflow = sess.run(batchnorm_out_tf)
 
-        np.testing.assert_array_almost_equal(
+          np.testing.assert_array_almost_equal(
             out_pond, out_tensorflow, decimal=1)
 
+    def test_channels_first():
```

Comment:
can you move these two out into their own methods on TestBatchnorm? that way if one of them fails in the future it's a bit more informative for the person running this file

Reply:
you can also look into using `pytest.mark.parametrize` for this if you're feeling ambitious! (but not necessary for merging)

Reply:
@jvmancuso Thanks for the review. I separated the tests. Tried looking at ```pytest.mark.parametrize``` but it seems to be not working great with ```unittest.TestCase``` subclasses -
https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes
https://stackoverflow.com/questions/18182251/does-pytest-parametrized-test-work-with-unittest-class-based-tests

Reply:
I see, in that case parameterization is probably beyond the scope of this PR. LGTM!